lib/plugins/azure/deploy/lib/deployFunctions.js

```diff
@@ -0,0 +1,147 @@
+
+fetch.Promise = BbPromise;
+
+module.exports = {
+  extractFunctionHandlers() {
+    this.deployedFunctions = [];
+
+    forEach(this.serverless.service.functions, (value, key) => {
+      if (key !== 'name_template') {
+        this.deployedFunctions.push({
+          name: key,
+          handler: value.handler,
+          exclude: value.exclude,
+          include: value.include,
+        });
+      }
+    });
+
+    return BbPromise.resolve();
+  },
+
+  zipFunctions() {
+    this.serverless.cli.log('Zipping functions...');
+    this.deployedFunctions.forEach((func, index) => {
+      // create a new zip instance so that old functions won't slip into the new zip archive
+      const zip = new Zip();
+      const servicePath = this.serverless.config.servicePath;
+
+      let exclude = func.exclude || [];
+
+      // add defaults for exclude
+      exclude = union(exclude, [
+        '.git',
+        '.gitignore',
+        '.DS_Store',
+        'serverless.yaml',
+        'serverless.env.yaml',
+      ]);
+
+      const include = func.include || [];
+      const handler = (last(func.handler.split('/'))).replace(/\\g/, '/');
+      const handlerFullPath = path.join(servicePath, handler);
+      const normalizedHandler = path.normalize(func.handler);
+
+      if (!handlerFullPath.endsWith(normalizedHandler)) {
+        throw new this.serverless.classes.Error(`The handler ${func.handler} was not found`);
+      }
+
+      const packageRoot = handlerFullPath.replace(normalizedHandler, '');
+
+      // Add function files
+      this.serverless.utils.walkDirSync(packageRoot).forEach((filePath) => {
+        const relativeFilePath = path.relative(packageRoot, filePath);
+        const shouldBeExcluded = exclude.some(sRegex => {
+          const regex = new RegExp(sRegex);
+          const matches = regex.exec(relativeFilePath);
+          return matches && matches.length > 0;
+        });
+        const shouldBeIncluded = include.some(sRegex => {
+          const regex = new RegExp(sRegex);
+          const matches = regex.exec(relativeFilePath);
+          return matches && matches.length > 0;
+        });
+
+        if (!shouldBeExcluded || shouldBeIncluded) {
+          zip.file(relativeFilePath, fs.readFileSync(filePath));
+        }
+      });
+
+      // Add functions.json
+      const functionsJson = this.serverless.service.resources.azure.functions[func.name];
+      zip.file('function.json', JSON.stringify(functionsJson));
+
+      // Create zipped data
+      const data = zip.generate({
+        type: 'nodebuffer',
+        compression: 'DEFLATE',
+        platform: process.platform,
+      });
+
+      this.deployedFunctions[index].zipFileData = data;
+    });
+
+    return BbPromise.resolve();
+  },
+
+  uploadZipFiles() {
+    this.serverless.cli.log('Uploading zip files to Azure...');
+    const targetWebsite = this.serverless.service.resources.azure.variables.sitename;
+    const user = process.env.AZURE_USERNAME;
```

Comment:
Should be able to get deployment credentials from ARM (.../Microsoft.Web/sites/*/config/publishingcredentials). Asking users to fetch this and set env variables isn't going to be smooth, from experience.